plugins/outputs/sql/sql_test.go

```diff
@@ -211,31 +211,35 @@ func TestMysqlIntegration(t *testing.T) {
 		testMetrics,
 	))
 
-	//dump the database
-	var rc int
-	rc, _, err = container.Exec([]string{
-		"bash",
-		"-c",
-		"mariadb-dump --user=" + username +
-			" --password=" + password +
-			" --compact --skip-opt " +
-			dbname +
-			" > /out/dump",
-	})
-	require.NoError(t, err)
-	require.Equal(t, 0, rc)
-	dumpfile := filepath.Join(outDir, "dump")
-	require.Eventually(t, func() bool {
-		_, err := os.Stat(dumpfile)
-		return !os.IsNotExist(err)
-	}, 5*time.Second, 10*time.Millisecond)
+	cases := []struct {
+		expectedFile string
+	}{
+		{"./testdata/mariadb/expected_metric_one.sql"},
+		{"./testdata/mariadb/expected_metric_two.sql"},
+		{"./testdata/mariadb/expected_metric_three.sql"},
+	}
+	for _, tc := range cases {
+		require.Eventually(t, func() bool {
+			rc, out, err := container.Exec([]string{
+				"bash",
+				"-c",
+				"mariadb-dump --user=" + username +
+					" --password=" + password +
+					" --compact --skip-opt " +
+					dbname,
+			})
+			require.NoError(t, err)
+			require.Equal(t, 0, rc)
```

Comment:
Do you really need this in the `Eventually` block as it will get executed every 10 milliseconds... Same for reading of the expected file...

Reply:
I too was hesitant to leave this in, but given the issues we have had with these integration tests I wanted to give this test every opportunity to wait for the database output. Some of the test failures made it seem that some of the data was just not present from the output, hence I left it in.

Reply:
As far as the 10 milliseconds, that is what I have been using for most of these require eventually blocks: 5 second timeout and 10 millisecond retry. I'm happy to change this, but my goal was let this finish fast on fast systems and still retry on the slow circle ci system

Reply:
Can we then please move at least the reading of the expectation file out and increase the polling time to e.g. one second?